src/main/java/com/github/checkstyle/regression/git/DiffParser.java

```diff
@@ -128,11 +120,55 @@ private static AbstractTreeIterator prepareTreeParser(Repository repository, Str
     }
 
     /**
+     * Gets the merge base of two commits.
+     * @param walk    the {@link RevWalk} for computing merge bases
+     * @param commitA the first commit to walk start with
+     * @param commitB the second commit to walk start with
+     * @return the merge base of two commits
+     * @throws IOException JGit library exception
+     */
+    private static RevCommit getBaseCommit(
+            RevWalk walk, RevCommit commitA, RevCommit commitB) throws IOException {
+        walk.reset();
+        walk.setRevFilter(RevFilter.MERGE_BASE);
+        walk.markStart(commitA);
+        walk.markStart(commitB);
+        return walk.next();
+    }
+
+    /**
+     * Creates a tree parser from a commit, to be used by diff command.
+     * @param walk   the {@link RevWalk} to parse the tree
+     * @param commit the commit to create tree parser from
+     * @return the tree parser
+     * @throws IOException JGit library exception
+     */
+    private static AbstractTreeIterator prepareTreeParser(RevWalk walk, RevCommit commit)
+            throws IOException {
+        final RevTree tree = walk.parseTree(commit.getTree().getId());
+        final CanonicalTreeParser returnValue;
+        returnValue = new CanonicalTreeParser();
+        returnValue.reset(walk.getObjectReader(), tree.getId());
+        return returnValue;
+    }
+
+    /**
      * Converts a {@link DiffEntry} to {@link GitChange} for the further use.
      * @param diffEntry the {@link DiffEntry} instance to be converted
      * @return the {@link GitChange} instance converted from the given {@link DiffEntry}
      */
     private static GitChange convertDiffEntryToGitChange(DiffEntry diffEntry) {
-        return new GitChange(diffEntry.getNewPath());
+        return ImmutableGitChange.builder()
+                .path(diffEntry.getNewPath())
+                .build();
+    }
+
+    /** A pair of tree parsers: feature tree parser and base tree parse. */
+    private static class TreeParserPair {
+        /** The tree parser of the head commit of feature branch. */
+        private AbstractTreeIterator featureTreeParser;
```

Comment:
I am fine with `prBranch*`. 
But `base*` and `master*` have different meaning in my current code. I named the head commit of master as `master*`, and the common ancestor of PR branch and master as `base*`. 

"the common ancestor of PR branch and master" has a formal related command [merge-base](https://git-scm.com/docs/git-merge-base) in Git and I found others are used to calling the common commit as `baseCommit`. Thats why I use `base*`.